packages/react/src/components/SelectUsersModal/SelectUsersModal.jsx

```diff
@@ -0,0 +1,266 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { ArrowRight16, ArrowLeft16 } from '@carbon/icons-react';
+import isEmpty from 'lodash/isEmpty';
+
+import ComposedModal from '../ComposedModal/ComposedModal';
+import HierarchyList from '../List/HierarchyList';
+import { Button } from '../..';
+import { settings } from '../../constants/Settings';
+
+const { iotPrefix } = settings;
+
+export const UserShape = PropTypes.shape({
+  id: PropTypes.string,
+  name: PropTypes.string,
+  email: PropTypes.string,
+  username: PropTypes.string,
+});
+
+export const GroupShape = PropTypes.shape({
+  id: PropTypes.string,
+  name: PropTypes.string,
+  groups: PropTypes.arrayOf(PropTypes.any),
+  users: PropTypes.arrayOf(UserShape),
+});
+
+export const propTypes = {
+  i18n: PropTypes.shape({
+    modalHeaderTitle: PropTypes.string,
+    modalHeaderLabel: PropTypes.string,
+    users: PropTypes.string,
+    available: PropTypes.string,
+    selected: PropTypes.string,
+    addUser: PropTypes.string,
+    removeUser: PropTypes.string,
+    recent: PropTypes.string,
+  }),
+  users: PropTypes.arrayOf(GroupShape).isRequired,
+  initialSelectedUsers: PropTypes.arrayOf(GroupShape).isRequired,
+
+  /** Should the dialog be open or not */
+  open: PropTypes.bool, // eslint-disable-line react/boolean-prop-naming
```

Comment:
the convention we have for props like this now is to ad the prefix `is` so `isOpen`. This will probably allow you to remove the disable eslint comment.